src/api/user.ts

```diff
@@ -0,0 +1,90 @@
+// Import Third-party Dependencies
+import * as httpie from "@myunisoft/httpie";
+
+// Import Internal Dependencies
+import * as utils from "../utils.js";
+
+interface NpmPaginated<T> {
+  total: number;
+  objects: T[];
+  urls: {
+    next: string;
+    prev: string;
+  }
+}
+
+interface NpmAvatars {
+  small: string;
+  medium: string;
+  large: string;
+}
+
+export interface NpmPackage {
+  id: number;
+  name: string;
+  description: string;
+  maintainers: string[];
+  version: string;
+}
+
+export interface PaginationParams {
+  perPage: number;
+  page: number;
+}
+
+export interface NpmUserProfile {
+  id: number;
+  name: string;
+  fullname?: string;
+  accounts: {
+    twitter?: string;
+    github?: string;
+  };
+  avatars: NpmAvatars
+  packages: NpmPaginated<NpmPackage>;
+  pagination: PaginationParams;
+}
+
+interface NpmWebUser {
+  scope: {
+    id: number;
+    parent: {
+      name: string;
+      avatars: NpmAvatars
+      resource: {
+        github?: string;
+        twitter?: string;
+        fullname?: string;
+      }
+    };
+  }
+  packages: NpmPaginated<NpmPackage>;
+  pagination: PaginationParams;
+}
+
+export async function user(
+  username: string,
+  { perPage, page }: Partial<PaginationParams> = { perPage: 25, page: 0 }
+): Promise<NpmUserProfile> {
+  if (typeof username !== "string" || username.length === 0) {
+    throw new TypeError("Argument `username` must be a non empty string");
+  }
+
+  const url = new URL(`~${username}?perPage=${perPage}&page=${page}`, utils.getNpmWeb());
```

Comment:
```suggestion
  const url = new URL(`~${username}`, utils.getNpmWeb());
  url.searchParams.set("perPage", perPage);
  url.searchParams.set("page", page);
```

Feel more readable/maintenable, wdyt?